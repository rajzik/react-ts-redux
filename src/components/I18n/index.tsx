import React, { ReactNode } from 'react';
import { IntlProvider } from 'react-intl';

import { Languages } from 'langTypes';
import messages from '@i18n';

interface I18nProps {
  language: Languages;
  children: ReactNode;
}

export default function I18n({ language, children }: I18nProps) {
  return (
    <IntlProvider locale={language} messages={{...messages[language]}}>
      {children}
    </IntlProvider>
  );
}
