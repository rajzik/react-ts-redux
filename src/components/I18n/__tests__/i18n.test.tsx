import React from 'react';
import { shallow } from 'enzyme';
import I18n from '..';
import renderer from 'react-test-renderer';
import { T } from '@components';

test('<I18n/>', () => {
  let i18n = renderer.create(
    <I18n language="cs">
      <div>
        <T>test</T>
      </div>
    </I18n>
  );
  expect(i18n.toJSON()).toMatchSnapshot();

  i18n = renderer.create(
    <I18n language="en">
      <div>
        <T>test</T>
      </div>
    </I18n>
  );

  expect(i18n.toJSON()).toMatchSnapshot();

});
