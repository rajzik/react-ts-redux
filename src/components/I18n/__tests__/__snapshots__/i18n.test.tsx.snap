// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<I18n/> 1`] = `
<div>
  <span>
    tohle nikdy nemen!
  </span>
</div>
`;

exports[`<I18n/> 2`] = `
<div>
  <span>
    This never change!
  </span>
</div>
`;
